nested/wafclient.json

```diff
@@ -0,0 +1,135 @@
+{
+    "$schema": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "moodleCommon": {
+            "metadata": {
+                "description": "Common Moodle values"
+            },
+            "type": "object"
+        },
+        "subnetIdWaf": {
+            "metadata": {
+                "description": "Azure resource ID of the subnet where this VM is to be deployed"
+            },
+            "type": "string"
+        },
+        "vmSetupParamsObj": {
+            "metadata": {
+                "description": "JSON-structured VM setup params that'll be injected to the VM (through cloud-init) and used by the custom script (setup_webserver.sh)"
+            },
+            "type": "object"
+        }
+    },
+    "resources": [
+        {
+            "type": "Microsoft.Network/networkInterfaces",
+            "apiVersion": "2017-10-01",
+            "location": "[parameters('moodleCommon').location]",
+            "name": "[parameters('moodleCommon').WafRestClientNicName]",
+            "properties": {
+                "ipConfigurations": [
+                    {
+                        "name": "ipcfg-waf-client",
+                        "properties": {
+                            "privateIPAllocationMethod": "Static",
+                            "privateIPAddress": "[parameters('moodleCommon').wafclientVmIP]",
+                            "subnet": {
+                                "id": "[parameters('subnetIdWaf')]"
+                            }
+                        }
+                    }
+                ]
+            },
+            "tags": {
+                "displayName": "WAF REST CLIENT NIC"
+            }
+        },
+        {
+            "type": "Microsoft.Compute/virtualMachines",
+            "apiVersion": "2017-03-30",
+            "dependsOn": [
+                "[concat('Microsoft.Network/networkInterfaces/', parameters('moodleCommon').WafRestClientNicName)]"
+            ],
+            "location": "[parameters('moodleCommon').location]",
+            "name": "[parameters('moodleCommon').wafclientVmName]",
```

Comment:
What is the purpose of this VM? Just to run the `install_waf.sh` once at the deployment time? Does this VM do anything after that? It doesn't look like so... Is this a good use of a dedicated VM, just to run a script once and keep it idle all the time thereafter? I think the script execution should be done in the already existing controller VM, and we shouldn't incur any more VM just for running a script once.